test/e2e-common.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# Copyright 2020 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script includes common functions for testing setup and teardown.
+source $(dirname $0)/../vendor/knative.dev/test-infra/scripts/e2e-tests.sh
+
+# Setup resources.
+function test_setup() {
+  echo ">> Setting up logging..."
+  # Install kail if needed.
+  if ! which kail > /dev/null; then
+    bash <( curl -sfL https://raw.githubusercontent.com/boz/kail/master/godownloader.sh) -b "$GOPATH/bin"
+  fi
+  # Capture all logs.
+  kail > ${ARTIFACTS}/k8s.log.txt &
+  local kail_pid=$!
+  # Clean up kail so it doesn't interfere with job shutting down
+  add_trap "kill $kail_pid || true" EXIT
+
+  # Setting up test resources.
+  echo ">> Publishing test images"
+  $(dirname $0)/upload-test-images.sh || fail_test "Error uploading test images"
+  echo ">> Creating test resources (test/config/)"
+  ko apply ${KO_FLAGS} -f test/config/ || return 1
+
+  # Bringing up controllers.
+  echo ">> Bringing up Istio"
+  echo ">> Running Istio CRD installer"
+  kubectl apply -f third_party/istio-latest/istio-crds.yaml || return 1
+  wait_until_batch_job_complete istio-system || return 1
+  echo ">> Running Istio"
+  kubectl apply -f third_party/istio-latest/istio-ci-no-mesh.yaml || return 1
```

Comment:
You know...  I'd be receptive to the idea of running all of the istio jobs presubmit in this repo 🤔